packages/shared_preferences/shared_preferences_android/test/shared_preferences_android_test.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:shared_preferences_android/shared_preferences_android.dart';
+import 'package:shared_preferences_platform_interface/method_channel_shared_preferences.dart';
+import 'package:shared_preferences_platform_interface/shared_preferences_platform_interface.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  group(MethodChannelSharedPreferencesStore, () {
+    const MethodChannel channel = MethodChannel(
+      'plugins.flutter.io/shared_preferences_android',
+    );
+
+    const Map<String, Object> kTestValues = <String, Object>{
+      'flutter.String': 'hello world',
+      'flutter.Bool': true,
+      'flutter.Int': 42,
+      'flutter.Double': 3.14159,
+      'flutter.StringList': <String>['foo', 'bar'],
+    };
+
+    late InMemorySharedPreferencesStore testData;
+
+    final List<MethodCall> log = <MethodCall>[];
+    late SharedPreferencesStorePlatform store;
+
+    setUp(() async {
+      testData = InMemorySharedPreferencesStore.empty();
+
+      channel.setMockMethodCallHandler((MethodCall methodCall) async {
+        log.add(methodCall);
+        if (methodCall.method == 'getAll') {
+          return await testData.getAll();
+        }
+        if (methodCall.method == 'remove') {
+          final String key = (methodCall.arguments['key'] as String?)!;
```

Comment:
Nit: you can write this slightly more simply as `methodCall.arguments['key']! as String`